ssl/statem/statem_lib.c

```diff
@@ -1049,7 +1049,7 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
                 const SSL_METHOD *method;
 
                 method = vent->smeth();
-                if (ssl_method_error(s, method) == 0) {
+                if (method != NULL && ssl_method_error(s, method) == 0) {
```

Comment:
I guess my comment wasn't good enough. It's smeth that can be NULL. You don't want to call smeth()